receiver/prometheusreceiver/metrics_receiver_non_numerical_test.go

```diff
@@ -0,0 +1,231 @@
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusreceiver
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/model/pdata"
+)
+
+var staleNaNsPage1 = `
+# HELP go_threads Number of OS threads created
+# TYPE go_threads gauge
+go_threads 19
+
+# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{method="post",code="200"} 100
+http_requests_total{method="post",code="400"} 5
+
+# HELP http_request_duration_seconds A histogram of the request duration.
+# TYPE http_request_duration_seconds histogram
+http_request_duration_seconds_bucket{le="0.05"} 1000
+http_request_duration_seconds_bucket{le="0.5"} 1500
+http_request_duration_seconds_bucket{le="1"} 2000
+http_request_duration_seconds_bucket{le="+Inf"} 2500
+http_request_duration_seconds_sum 5000
+http_request_duration_seconds_count 2500
+
+# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
+# TYPE rpc_duration_seconds summary
+rpc_duration_seconds{quantile="0.01"} 1
+rpc_duration_seconds{quantile="0.9"} 5
+rpc_duration_seconds{quantile="0.99"} 8
+rpc_duration_seconds_sum 5000
+rpc_duration_seconds_count 1000
+`
+
+var (
+	totalScrapes = 10
+)
+
+// TestStaleNaNs validates that staleness marker gets generated when the timeseries is no longer present
+func TestStaleNaNs(t *testing.T) {
+	mockResponses := make([]mockPrometheusResponse, 0)
+	for i := 0; i < totalScrapes; i++ {
+		if i%2 == 0 {
+			mockResponses = append(mockResponses, mockPrometheusResponse{
+				code: 200,
+				data: staleNaNsPage1,
+			})
+		} else {
+			mockResponses = append(mockResponses, mockPrometheusResponse{
+				code: 500,
+				data: "",
+			})
+		}
+	}
+	targets := []*testData{
+		{
+			name:            "target1",
+			pages:           mockResponses,
+			validateFunc:    verifyStaleNaNs,
+			validateScrapes: true,
+		},
+	}
+	testComponent(t, targets, false, "")
+}
+
+func verifyStaleNaNs(t *testing.T, td *testData, resourceMetrics []*pdata.ResourceMetrics) {
+	verifyNumTotalScrapeResults(t, td, resourceMetrics)
+	metrics1 := resourceMetrics[0].InstrumentationLibraryMetrics().At(0).Metrics()
+	ts1 := getTS(metrics1)
+	for i := 0; i < totalScrapes; i++ {
+		if i%2 == 0 {
+			verifySuccessfulScrape(t, td, resourceMetrics[i], &ts1, i+1)
+		} else {
+			verifyFirstFailedScrape(t, td, resourceMetrics[i], &ts1, i+1)
+		}
+	}
+}
+
+func verifySuccessfulScrape(t *testing.T, td *testData, resourceMetric *pdata.ResourceMetrics, startTimestamp *pdata.Timestamp, iteration int) {
```

Comment:
This needs a more descriptive name since it is applicable to a specific test.

Reply:
@Aneurysm9 thanks. Updated name from  `verifySuccessfulScrape` to `verifyStaleNaNPage1SuccessfulScrape` , `verifyFirstFailedScrape` to  `verifyStaleNanPage1FirstFailedScrape`.